tests/test_monitor.py

```diff
@@ -20,6 +20,59 @@ def test_add_monitor(hlwm):
     assert hlwm.get_attr('monitors.1.name') == 'monitor2'
 
 
+def test_cannot_add_monitor_without_free_tag(hlwm):
+    call = hlwm.call('add_monitor', '800x600+40+40', check=False)
+
+    assert call.returncode != 0
+    assert call.stderr.endswith(' not enough free tags\n')
```

Comment:
I'm not sure how important the error message looks like, but it's important that the command fails (we also don't care about the returncode as long as it is non-zero here). If you agree, you can boil the test down to:
```
hlwm.fails('add_monitor', '800x600+40+40')
assert hlwm.get_attr('monitors.count') == '1'
```

Reply:
If there is only one error case it doesn't matter, I assume the rationale behind this approach is to ensure the right error is given.

Reply:
For tests like these I strongly recommend verifying the failure cause. It's easy to mask errors in the code or simply mistakes in the test if you just expect it to fail. For example, if you have a typo in `'add_monitor'` in the test, it will be green.

That said, I would prefer checking the error message in a more precise fashion than string matching. I shortly look into checking the actual return code, but for that I'd have to evaluate `ipc-protocol.h`, which doesn't even hard-code them.

As an intermediate solution, we could duplicate the exit code definitions in the tests.